dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetFactoryIntegrationTest.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    /**
+     * Method to test: {@link FileAssetFactory#findFileAssetsByFolderInDB(Folder, User, boolean)}
+     * When: Create one Folder with two files each one, grant permission to anonymous role
+     * Should: return two files
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void shouldWorkWithAnonymous() throws IOException, DotDataException, DotSecurityException {
+        final Role anonymousRole = APILocator.getRoleAPI().loadCMSAnonymousRole();
+        final User user = new UserDataGen().nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Folder folder = new FolderDataGen().site(host).nextPersisted();
+        final Contentlet fileAsset1 = createFileAsset(folder, "text1", ".txt");
+        final Contentlet fileAsset2 = createFileAsset(folder, "text2", ".txt");
+
+        ContentletDataGen.publish(fileAsset1);
+        ContentletDataGen.publish(fileAsset2);
+
+        this.addPermission(anonymousRole, folder, fileAsset1, fileAsset2);
+        final List<Contentlet> files = fileAssetFactory.findFileAssetsByFolderInDB(folder, user, true);
+
+        assertEquals(2, files.size());
+
+        final List<String> filesInodes = files.stream().map((file) -> file.getInode()).collect(Collectors.toList());
+        assertTrue(filesInodes.contains(fileAsset1.getInode()));
+        assertTrue(filesInodes.contains(fileAsset2.getInode()));
+    }
+
+    /**
+     * Method to test: {@link FileAssetFactory#findFileAssetsByFolderInDB(Folder, User, boolean)}
+     * When: Create one Folder without files
+     * Should: return a empty list
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void shouldReturnEmptyList() throws DotDataException, DotSecurityException {
+        final Role backEndUserRole = APILocator.getRoleAPI().loadBackEndUserRole();
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+        final Folder folder = new FolderDataGen().site(host).nextPersisted();
+
+        this.addPermission(role, folder);
+        final List<Contentlet> files = fileAssetFactory.findFileAssetsByFolderInDB(folder, user, false);
+
+        assertTrue(files.isEmpty());
+    }
+
+    /**
+     * Method to test: {@link FileAssetFactory#findFileAssetsByFolderInDB(Folder, User, boolean)}
+     * When: Create a Folder and not add permission
+     * Should: throw a {@link DotSecurityException}
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldThorDotSecurityException() throws DotDataException, DotSecurityException {
```

Comment:
I would rename this method. Even when there are comments that explain the test, the name itself doesn't say something really helpful as the annotation `@Test(expected = DotSecurityException.class)`  says exactly that

Reply:
done https://github.com/dotCMS/core/pull/18225/commits/a14571cd0fa1e43bd5e21bfd4b25d77057a712f1#diff-b1dc6015d0a0a590385cffd31ab8418eR175